### ISSUE TITLE
Improve Add/Select Link widget

### ISSUE THREAD

**Issue by sneridagh** [plone/volto#4303]
## Abstract

Improve the overall UX for the "add a link to a internal content or external URL".

## Motivation

We already have one `AddLinkForm` that we use in a couple of scenarios to link content (in Slate editor and in the Image block and component alike. The idea is to improve and unify the experience across all the elements that has this interaction.

I know we had something similar before, but I think that now is more relevant than ever. I seems that it's a pattern that is well accepted and present in Google products, Confluence, WP, etc...

<img width="512" alt="image" src="https://user-images.githubusercontent.com/486927/214250274-37f62f99-c6b7-4fd6-a349-5c5b75bf7ba9.png">

<img width="845" alt="image" src="https://user-images.githubusercontent.com/486927/214283994-0ca21bbf-4994-44c3-a4c5-06acd1810143.png">

## Assumptions

There are no assumptions.

## Proposal & Implementation

We need to define the final look of the widget and include the feature of searching "in place". 

- Decide on which elements to show by default
  - The content inside the folder in which I'm in?
  - The most recent content the current user has created?
  - Fallback in case none of the above return results
- How the link is "saved"/"commited"
- Edit the link
- Remove the link
- Check corner cases

## Deliverables

A new component that provides this features and can be reused in other scenarios.

## Participants

Víctor Fernández de Alba 
... help is welcomed!

**Comment by akshatgarg12** [comment]
I would like to assist in this @sneridagh

**Comment by saadabban76** [comment]
can I work on this issue ? @akshatgarg12

**Comment by stevepiercy** [comment]
@akshatgarg12 @saadabban76 this issue is a PLIP, not an issue to be worked on at this time. A PLIP must be discussed, reviewed, and approved with a task list of deliverables by the project team before any work can be performed. Please read https://5.docs.plone.org/develop/coredev/docs/plips.html for details.

**Comment by saadabban76** [comment]
> @akshatgarg12 @saadabban76 this issue is a PLIP, not an issue to be worked on at this time. A PLIP must be discussed, reviewed, and approved with a task list of deliverables by the project team before any work can be performed. Please read https://5.docs.plone.org/develop/coredev/docs/plips.html for details.

Thanks for the info, I didn't know about this 👍

**Comment by akshatgarg12** [comment]
@sneridagh We already have a `branchIcon` clicking on which allows us to choose image from other pages, should the link component show these images/ content from other pages in a dropdown, along with recently added urls and content in the site. 
![Screenshot 2023-03-19 at 3 59 30 PM](https://user-images.githubusercontent.com/60404253/226170187-ef42350b-3d1b-45b5-a500-82c8a7a5b57f.png)
![Screenshot 2023-03-19 at 4 14 52 PM](https://user-images.githubusercontent.com/60404253/226170212-1b22fe2b-8c06-46e7-892e-0bff7c60fa6c.png)

**Comment by Darkcoder011** [comment]
can i take this is issue. I work on React in  more than 4 project 
This the template that was created in react ans tailwind 

![Screenshot 2023-03-19 at 21-06-23 Ghost Gatsby Starter - Ghost   Gatsby](https://user-images.githubusercontent.com/107787867/226244427-d7ad8a2d-9a80-4cf8-9ace-f52a2a639eb9.png)

**Comment by stevepiercy** [comment]
@Darkcoder011 https://github.com/plone/volto/issues/4303#issuecomment-1407503252

**Comment by nileshgulia1** [comment]
@Darkcoder011 Please don't post irrelevant comments in the github issue tracker. If you want to contribute and make an impact follow: https://6.docs.plone.org/volto/developer-guidelines/contributing.html?highlight=contributing#contributing-to-volto

**Comment by priyansu12** [comment]
I would like to contribute on this issue.

**Comment by stevepiercy** [comment]
@priyansu12 see https://github.com/plone/volto/issues/4303#issuecomment-1407503252. Deleting your comment as irrelevant.

**Comment by bhimanshu1** [comment]
@sneridagh can I do this using the other web applications like css and js

**Comment by sneridagh** [comment]
@bhimanshu1 what do you mean? Are you interested in work in this issue?

**Comment by bhimanshu1** [comment]
@sneridagh yes i am interested in the issue

**Comment by shubham-200315** [comment]
Hello @sneridagh i would like to work on this issue.

**Comment by stevepiercy** [comment]
@bhimanshu1 @shubham-200315 please read and follow [Contributing to Plone](https://6.docs.plone.org/contributing/index.html) and [First-time contributors](https://6.docs.plone.org/contributing/first-time.html).

**Comment by stevepiercy** [comment]
@sneridagh can we include the ability to set link anchors within a page? See https://github.com/plone/volto/issues/593 from 2019.

**Comment by sneridagh** [comment]
@stevepiercy yes. This would be a great addition indeed, but opens a whole different branch of features to implement :/

**Comment by tisto** [comment]
@plone/volto-team what's the status of this PLIP? We are wrapping up the planning for Plone 6.2 which will happen in Nov/Dec 2025. If we want to ship this with Volto 19 and Plone 6.2 we need to push this and merge it within the next few weeks. Do we have the capacity to do so? Otherwise, we would have to schedule this for Plone 6.3 and possibly Volto 20 (or 21).

**Comment by davisagli** [comment]
@dobri1408 worked on this at the Beethoven Sprint, that work is in https://github.com/plone/volto/pull/7120. @pnicolli reviewed it and left several comments that have no response. @dobri1408 @avoinea is this something you can finish up, or do we need to find someone else to take over?

**Comment by dobri1408** [comment]
Hi! I will try to finish it this week!

**Comment by tisto** [comment]
@dobri1408 awesome! do you see a chance to finish your work on this in the next weeks? I plan to mention this in our state-of-Plone keynote, and we need to decide if we plan this for Plone 6.2 (when we can finish it in the next few weeks around Plone conf) or if we postpone it to 6.3.

**Comment by dobri1408** [comment]
@tisto  Ok

**Comment by tisto** [comment]
@dobri1408 any update on this? ATM I would schedule this rather for Plone 6.3.

**Comment by tisto** [comment]
@dobri1408 @sneridagh any update on this? Is this something we want to do for Plone 6.3 or shall we drop this altogether.
